rewrite-core/src/main/java/org/openrewrite/RecipeScheduler.java

```diff
@@ -35,41 +35,46 @@
 import java.util.concurrent.Callable;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.function.BiFunction;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
 
 import static java.util.Collections.*;
 import static java.util.Objects.requireNonNull;
 import static org.openrewrite.Recipe.PANIC;
 import static org.openrewrite.RecipeSchedulerUtils.addRecipesThatMadeChanges;
 import static org.openrewrite.RecipeSchedulerUtils.handleUncaughtException;
-import static org.openrewrite.Tree.randomId;
 
+/**
+ * The scheduler is responsible for executing a {@link Recipe} full lifecycle and
+ * reporting a {@link RecipeRun} result.
+ */
 public interface RecipeScheduler {
-    default <T> List<T> mapAsync(List<T> input, BiFunction<T, Integer, T> mapFn) {
+    default <T> List<T> mapAsync(List<T> input, UnaryOperator<T> mapFn) {
```

Comment:
Changing this signature is a breaking change for our saas. I would rather not spring it on them without good reason. I am fine with most of the changes in this PR but don't want to change the public method signatures right now